src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,10 +22,58 @@ Retrieve a project's configuration.
+| ------- | ----------- | ----------------------------------------------------------------------------------------- |
+| rollout | Rollout     | An object which contains instructions for rolling out a variant.                          |
+| rules   | array[Rule] | An array of rules which all must evaluate to true in order to return the variant's value. |
+| value   | any         | The value of the feature if the variant is successfully evaluated.                        |
+
+When evaluating a set of rules and varaints each rule is a logical `AND` while each variant is a logical `OR`. Failure to evaluate one variant does not prevent another variant from successfully evaluating. Failure to evaluate one rule does invalidate the remaining rules and the evaluation proceedure can eagerly terminate.
+
+**Rollout Object**
+
+| Field      | Type   | Description                                                 |
+| ---------- | ------ | ----------------------------------------------------------- |
+| percentage | number | The percentage of requests which should evaluate to true.   |
+| seed       | string | A SHA1 hash used to seed a random number generator.         |
+| target     | string | The name of a property contained within the context object. |
+
+The rollout cohort a session is bucketed into is deterministic some long as the `target` value is static. The `target` value controls the bucket a session is placed into. The `seed` value provides randomization _between features_ such that if a session is successfully opted into feature `A` they will not necesssarily be opted into feature `B` even if those features share the same rollout percentage. The `percentage` value controls the number of buckets which evaluate to `true`.
+
+**Rule Object**
+
+| Field    | Type                                        | Description                                                             |
+| -------- | ------------------------------------------- | ----------------------------------------------------------------------- |
+| operator | Operator                                    | A conditional operator that evaluates the value against the target.     |
+| property | string                                      | A context object property name which contains the value of the operand. |
+| value    | union[string, int, bool, null, array[self]] | The value being compared against.                                       |
+
+Rules contain evaluation and extraction instructions. The context object provided must contain the key defined on the `property` attribute. If the SDK can not find the `property` within the context object the evaluation fails and the variant is skipped. The `operator` defines how the context value is compared against the literal provided by the configuration on the `value` attribute. If the value provided in the context object can not be compared against the value provided in the configuration the evaluation fails and the variant is skipped.
```

Comment:
IMO we should also write down that we're comparing against json values, so `null` is acceptable, `None` (python) and `undefined` (js) are not. 
Similarly, when comparing values we should be specific about equality. Falsey values and type coercion come to mind. Depending on type coercion, a missing properly might match against an expected falsey value and pass the condition, or it might never match and therefore fail the condition. 
But things like `?context={age: false}` with `condition=['age', '<', 21]` need to be consistent in each SDK. Certainly if open feature handles it then we're done here and I'll go read about it!